frontend/src/lib/components/TaxonomicPopover/TaxonomicPopover.tsx

```diff
@@ -107,6 +53,16 @@ export function LemonTaxonomicPopover<ValueType extends TaxonomicFilterValue = T
 
     const isClearButtonShown = allowClear && !!localValue
 
+    const internalButtonProps = buttonProps as LemonButtonProps
+    internalButtonProps.children = localValue ? (
+        <span>{renderValue?.(localValue) ?? localValue}</span>
+    ) : (
+        <span className={placeholderClass ?? 'text-muted'}>{placeholder}</span>
+    )
+    internalButtonProps.onClick = () => setVisible(!visible)
+    internalButtonProps.status = 'stealth'
+    internalButtonProps.type = 'secondary'
```

Comment:
Naming Nit: I'd call `buttonProps` something like `restButtonProps` to indicate it's a cloned object with partial keys, and not a prop that was passed in, which we're now mutating.